api/src/main/java/org/apache/iceberg/PartitionSpec.java

```diff
@@ -427,13 +429,21 @@ private void checkForRedundantPartitions(PartitionField field) {
       dedupFields.put(dedupKey, field);
     }
 
+    public Builder caseSensitive(boolean sensitive) {
+      this.caseSensitive = sensitive;
+      return this;
```

Comment:
Can we just have a void `caseInsensitiveFields` API? The default is false (preserving the original behavior), and then when this API gets called it sets it to true.

Reply:
> current `caseSensitive(boolean)` seems more consistent with existing code, like `Scan` interface.

That is exactly why I implemented it this way